examples/video_list_to_csv.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import logging
+import sys
+import os
+import csv
+import time
+
+import jwplatform
+
+
+def make_csv(api_key, api_secret, path_to_csv=None, result_limit=1000, **kwargs):
+    """
+    Function which fetches a video library and writes each video_objects Metadata to CSV. Useful for CMS systems.
+
+    :param api_key: <string> JWPlatform api-key
+    :param api_secret: <string> JWPlatform shared-secret
+    :param path_to_csv: <string> Local system path to desired CSV. Default will be within current working directory.
+    :param result_limit: <int> Number of video results returned in response. (Suggested to leave at default of 1000)
+    :param kwargs: Arguments conforming to standards found @ https://developer.jwplayer.com/jw-platform/reference/v1
+    :return: <dict> Dict which represents the JSON response.
+    """
+
+    path_to_csv = path_to_csv or os.path.join(os.getcwd(), 'video_list.csv')
+    timeout = 2
+    offset = 0
+    videos = list()
+
+    jwplatform_client = jwplatform.Client(api_key, api_secret)
+    logging.info("Querying for video list.")
+
+    while True:
+        try:
+            response = jwplatform_client.videos.list(result_limit=result_limit,
+                                                     result_offest=offset,
+                                                     **kwargs)
+        except jwplatform.errors.JWPlatformRateLimitExceededError:
+            logging.error("Encountered rate limiting error. Backing off on request time.")
```

Comment:
i think you need a "max_retry" counter, to count how many times you retried.  If we are doing it with "seconds", I would recommend:

    # line 26 or so...
    max_retries = 3
    timeout_in_seconds = 1

And change line 33 to a `for i = 0; i < max_retries; i++`...

That's one option.  We could leave the `while True`, but we'd have to be careful to break out of all cases.

The way it is written now I believe it would pause-n-loop forever.